### ISSUE TITLE
Add confirmation message when clicking external links to `more-links`

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5335]
### Why this would be helpful/solve a problem

It's nice to have the option of clickable external links, but not every external link is safe. If you click on one of these links by accident, it can lead to malicious downloads, scams, or other similar situations that you don't want to be in.

### What this change does

This would add the option to enable a confirmation message that comes up whenever you're about to be redirected to an external website by clicking a link that `more-links` linkified.

### Possible alternatives

There is already a way to avoid accidental clicks. Since Scratch doesn't linkify external links by default (as you know by using this addon), it is possible to visit these websites from the browser's right-click context menu. You can try it here (assuming you're using a browser that has that option in its context menu, such as Chrome) by selecting the text below, right-clicking it, and choosing "Go to (url)".
`https://github.com/ScratchAddons/ScratchAddons`

With the addon turned off, using this trick makes it so that it takes two clicks to go to external websites. It still helps you avoid accidents as described earlier, but also is a bit more involved and tedious than just clicking and confirming. Personally, I think having an option (maybe on by default) to be asked before taking external links is a better solution.

### Demonstration(s) and additional context

Popup would look like:
![Are you sure you want to visit this website? Some links may lead to unsafe content like scams or viruses.](https://user-images.githubusercontent.com/106490990/200469981-a19685b0-1e90-46a9-b09b-02fa57960f4b.png)

**Comment by WorldLanguages** [comment]
Not sure if this makes much sense

**Comment by BroJac5246** [comment]
> Not sure if this makes much sense

As long as it's an option, I don't see why not.

**Comment by DNin01** [comment]
I have mixed feelings. At first glance, this doesn’t seem very necessary, and if people are posting inappropriate links, the posts can be reported and taken down. But when you think about it more, Scratch is a kids website and kids may not have as much experience with staying safe online, and it’s certainly possible for a kid to have Scratch Addons.

Even Minecraft has a warning message like this.

Let me ask, what’s our target audience?

**Comment by Secret-chest** [comment]
> Even Minecraft has a warning message like this.

Even Discord does.

However, the link text is always the URL, so you can see it if you're careful.

**Comment by BroJac5246** [comment]
> I have mixed feelings. At first glance, this doesn’t seem very necessary, and if people are posting inappropriate links, the posts can be reported and taken down.

I agree with this. Scratch is well-moderated, so this probably isn't necessary.

> But when you think about it more, Scratch is a kids website and kids may not have as much experience with staying safe online, and it’s certainly possible for a kid to have Scratch Addons.

I don't think we're targeting, like, 8-year-olds, but that doesn't mean they don't use this extension. And it can't hurt, can it? But the question is, would anyone turn it on? Probably not, and I don't think anyone would like it being on by default.

**Comment by WorldLanguages** [comment]
> Let me ask, what’s our target audience?

Well, if they have this addon enabled, then they already installed an extension in their browser and gave it perms such as accessing all data on `scratch.mit.edu`. In comparison, clicking a link doesn't sound very dangerous.

**Comment by DNin01** [comment]
> > Let me ask, what’s our target audience?
> 
> 
> 
> Well, if they have this addon enabled, then they already installed an extension in their browser and gave it perms such as accessing all data on `scratch.mit.edu`. In comparison, clicking a link doesn't sound very dangerous.

What if a parent or teacher installed Scratch Addons? Does that happen often?

**Comment by WorldLanguages** [comment]
> Does that happen often?

We have no way to know, really.

Maybe the more links addon could have something similar to the "dangerous" tag but with a custom warning message, when enabling the addon. But I don't think it makes sense to prompt the user when actually clicking links.

**Comment by BroJac5246** [comment]
> > Does that happen often?
> 
> We have no way to know, really.
> 
> Maybe the more links addon could have something similar to the "dangerous" tag but with a custom warning message, when enabling the addon. But I don't think it makes sense to prompt the user when actually clicking links.

We could add a notice or warning in the addon's settings.

**Comment by DNin01** [comment]
Another thing that might be noteworthy for some users of this addon is that just because you're using this addon doesn't mean others will be able to follow the external links that you can. If someone got in the habit of sharing external links, they might be confused as to why their recipients/viewers can't click the links.

**Comment by BroJac5246** [comment]
> Another thing that might be noteworthy for some users of this addon is that just because you're using this addon doesn't mean others will be able to follow the external links that you can. If someone got in the habit of sharing external links, they might be confused as to why their recipients/viewers can't click the links.

Pretty much everyone knows a URL when they see one, so that won't be much of a problem, just an annoyance.

**Comment by Samq64** [comment]
Maybe this could be part of the confirmations addon?
